File: php/testcode/hardcodedcreds_043.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_hardcoded_gcp_creds_path
function hardcodedcreds043(BenchmarkRequest $req): BenchmarkResponse {
    putenv('GOOGLE_APPLICATION_CREDENTIALS=' . getenv('GCP_CREDS_PATH')); // vuln-code-snippet safe-line php_hardcoded_gcp_creds_path
    $project = $req->param('project');
    $scope = 'https://www.googleapis.com/auth/cloud-platform';
    return BenchmarkResponse::ok('gcp auth configured for project: ' . $project);
}
// vuln-code-snippet end php_hardcoded_gcp_creds_path

File: ruby/testcode/hardcodedcreds_043.rb
require 'net/smtp'
require_relative 'shared'

# vuln-code-snippet start ruby_hardcoded_encrypted_creds_file
def send_email_safe_handler(req)
  smtp_pass = Rails.application.credentials.smtp_pass  # vuln-code-snippet safe-line ruby_hardcoded_encrypted_creds_file
  Net::SMTP.start('smtp.example.com', 587, 'localhost', 'user@example.com', smtp_pass, :login)
  BenchmarkResponse.ok('email sent')
end
# vuln-code-snippet end ruby_hardcoded_encrypted_creds_file
